project-service/api.md

```diff
@@ -0,0 +1,594 @@
+    * Content: `{ error: "You are unauthorized to make this request."}`
+
+### Add User to Project
+
+Add a user to a project
+
+- URL
+  + `/projects/users`
+- Method
+  + `POST`
+- URL Params
+  + None
+- Data Params
+  + Required
+    * `projectId=[number]`
+    * `userId=[number]`
+- Success Response
+  + Code: `204 NO CONTENT`
+- Error Response
+  + Invalid Project
+    * Code: `404 NOT FOUND`
+    * Content: `{ error: "Project doesn't exist." }`
+  + Invalid User
+    * Code: `404 NOT FOUND`
+    * Content: `{ error: "User doesn't exist." }`
+  + Not Authenticated
+    * Code: `401 UNAUTHORIZED`
+    * Content: `{ error: "You are unauthorized to make this request."}`
+
+### Remove User from Project
+
+Remove a user to a project
+
+- URL
+  + `/projects/users`
+- Method
+  + `DELETE`
+- URL Params
+  + None
+- Data Params
+  + Required
+    * `projectId=[number]`
+    * `userId=[number]`
+- Success Response
+  + Code: `204 NO CONTENT`
+- Error Response
+  + Invalid Project
+    * Code: `404 NOT FOUND`
+    * Content: `{ error: "Project doesn't exist." }`
+  + Invalid User
+    * Code: `404 NOT FOUND`
+    * Content: `{ error: "User doesn't exist." }`
+  + Not Authenticated
+    * Code: `401 UNAUTHORIZED`
+    * Content: `{ error: "You are unauthorized to make this request."}`
+
+## Sprints
+
+### Fetch Sprints
+
+Get all existing sprints for a project
+
+- URL
+  + `/sprints`
+- Method
+  + `GET`
+- URL Params
+  + Required
+    * `projectId=[number]`
+- Data Params
+  + None
+- Success Response
+  + Code: `200 OK`
+  + Content:
+
+```
+[
+  {
+    id: 1,
+    name: 'mvp'
+    status: 'Started',
+    startDate: [date],
+    endDate: [date]
+  }, {
+    id: 2,
+    name: 'profit',
+    status: 'Not Started',
+    startDate: [date],
+    endDate: [date]
+  }
+]
+```
+
+- Error Response
+  + Invalid Project
+    * Code: `404 NOT FOUND`
+    * Content: `{ error: "Project doesn't exist." }`
+  + Not Authenticated
+    * Code: `401 UNAUTHORIZED`
+    * Content: `{ error: "You are unauthorized to make this request."}`
+
+### Create Sprint
+
+Create a new sprint for a project
+
+- URL
+  + `/sprints`
+- Method
+  + `POST`
+- URL Params
+  + Required
+    * `projectId=[number]`
+- Data Params
+  + Required
+    * `name=[string]`
+    * `status=[string]`
+    * `startDate=[date]`
+    * `endDate=[date]`
+- Success Response
+  + Code: `201 CREATED`
+  + Content:
+
+```
+{
+  id: 1,
+  name: 'extra features',
+  status: 'Not Started',
+  startDate: [date],
+  endDate: [date],
+  createdAt: [date]
+}
+```
+
+- Error Response
+  + Invalid Project
+    * Code: `404 NOT FOUND`
+    * Content: `{ error: "Project doesn't exist." }`
+  + Not Authenticated
+    * Code: `401 UNAUTHORIZED`
+    * Content: `{ error: "You are unauthorized to make this request."}`
+
+### Edit Sprint
+
+Edit details of a sprint
+
+- URL
+  + `/sprints`
+- Method
+  + `PUT`
+- URL Params
+  + Required
+    * `sprintId=[number]`
+- Data Params
+  + Optional
```

Comment:
Ok, important part: we need to be able to add tasks to a sprint. And I guess it should be allowed as a bulk action, so the backend should expect either one or many tasks sent from the frontend to be added to the sprint